### ISSUE TITLE
Global macros

### ISSUE THREAD

**Issue by virustotalop** [ClubObsidian/DynamicGui#102]
Should global macros be in the config file, one separate file or should we have a folder where users can put numerous different macro files. Global macros when then get `compiled` into guis on reload/start time.

**Comment by haloflooder** [comment]
I think having a global macro folder with a bunch of files would be a better choice. But it would be better to "load in" macros that way you know what you're working with. Similar to how it works with OL where you do `global.get("thing")`.

### Example
```
Global macro folder contents:
- serverinfo.yml
- crateinfo.yml
- emotepixlprices.yml
```
```yaml
# gui config
load-macros:
- serverinfo
- emotepixlprices
```

**Comment by virustotalop** [comment]
Hmm alright that can be done. I'll start implementing that in a few hours or so.

**Comment by virustotalop** [comment]
How should macros be defined in their file? Should there be a list of macros where it just parses every root key or should it just parse one root key that is a defined name?

**Comment by haloflooder** [comment]
So everything in the "macros" root
```yaml
macros:
  goldlitedesc: "This is a very cheap thing"
  '%gold-lite-price%': 1
  goldehdesc: "This is a bit more expensive"
  `%gold-eh-price%`: 10
  goldheavydesc: "You must be hella rich"
  `%gold-heavy-price%`: 1000
```
or just use root
```yaml
goldlitedesc: "This is a very cheap thing"
'%gold-lite-price%': 1
goldehdesc: "This is a bit more expensive"
`%gold-eh-price%`: 10
goldheavydesc: "You must be hella rich"
`%gold-heavy-price%`: 1000
```
or root is only used as an organizer and the content in the root is what actually matters
```yaml
goldlitemacros: 
  goldlitedesc: "This is a very cheap thing"
  '%gold-lite-price%': 1

goldehmacros:
  goldehdesc: "This is a bit more expensive"
  `%gold-eh-price%`: 10

goldheavymacros:
  goldheavydesc: "You must be hella rich"
  `%gold-heavy-price%`: 1000
```

**Comment by virustotalop** [comment]
Implemented as of 7d771af943feb1136154b61eba5ab6a3e3250f90
